files/initramfs-tools/union-mount.j2

```diff
@@ -47,9 +47,17 @@ mount -n -o lowerdir=${rootmnt},upperdir=${rootmnt}/host/$image_dir/rw,workdir=$
 [ -b ${ROOT} ] || mdev -s
 ## Mount the raw partition again
 mount ${ROOT} ${rootmnt}/host
-## Mount the working directory of docker engine in the raw partition, bypass the overlay
+
 mkdir -p ${rootmnt}/var/lib/docker
-mount --bind ${rootmnt}/host/$image_dir/{{ DOCKERFS_DIR }} ${rootmnt}/var/lib/docker
+if [ -f ${rootmnt}/host/$image_dir/{{ FILESYSTEM_DOCKERFS }} ]; then
+    ## mount 1024M tmpfs and extract docker into it
+    mount -t tmpfs -o rw,nodev,size=1024M tmpfs ${rootmnt}/var/lib/docker
```

Comment:
1024M is currently hardcoded value, will need to figure an automatic way to cacluate the tmpfs size.

Reply:
```
if [ -d ${rootmnt}/host/$image_dir/{{ DOCKERFS_DIR }} ]; then
    ## Mount the working directory of docker engine in the raw partition, bypass the overlay
    mount --bind ${rootmnt}/host/$image_dir/{{ DOCKERFS_DIR }} ${rootmnt}/var/lib/docker
elif [ -f ${rootmnt}/host/$image_dir/{{ FILESYSTEM_DOCKERFS }} ]; then
    ## mount 1024M tmpfs and extract docker into it
    mount -t tmpfs -o rw,nodev,size=1024M tmpfs ${rootmnt}/var/lib/docker
    tar xz --numeric-owner --warning=no-timestamp -f ${rootmnt}/host/$image_dir/{{ FILESYSTEM_DOCKERFS }} -C ${rootmnt}/var/lib/docker
fi
```

---
In reply to: [310374338](https://github.com/Azure/sonic-buildimage/pull/3279#discussion_r310374338) [](ancestors = 310374338)

Reply:
prefer to check the dockerfs.tar.gz file, the directory could be accidentally created.